File: src/main/java/com/staymate/service/HostelService.java
package com.staymate.service;

import java.util.List;

import com.staymate.dto.HostelRegistrationDTO;
import com.staymate.entity.Hostel;

public interface HostelService {

    Hostel registerHostel(HostelRegistrationDTO dto);

    List<Hostel> getPendingHostels();

    Hostel approveHostel(Long hostelId);

    Hostel rejectHostel(Long hostelId);
    
    List<Hostel> getApprovedHostels();

}

File: frontend/src/services/hostelService.js
import api from "./api";

const hostelService = {

    registerHostel: (hostel) =>
        api.post("/hostels/register", hostel),

    getPendingHostels: () =>
        api.get("/hostels/pending"),

    getApprovedHostels: () =>
        api.get("/hostels/approved"),

    approveHostel: (id) =>
        api.put(`/hostels/approve/${id}`),

    rejectHostel: (id) =>
        api.delete(`/hostels/reject/${id}`)

};

export default hostelService;